lib/gradebook.rb

```diff
@@ -1 +1,35 @@
+class GradeBook
+  attr_reader :instructor, :courses
 
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+  end
+
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    students_by_course = {}
+    @courses.each do |course|
+      students_by_course[course] = course.students
+    end
+    students_by_course
+  end
+
+  def students_below(threshold)
+    all_students = @courses.map { |course| course.students }.flatten
```

Comment:
```suggestion
    all_students = @courses.flat_map { |course| course.students }
```

The way you have it is totally fine. You can also use `flat_map` which streamlines things a bit since we don't need to chain methods.